src/Lib/Canvas.vala

```diff
@@ -24,14 +24,13 @@
 
 public class Akira.Lib.Canvas : Goo.Canvas {
     public weak Akira.Window window { get; construct; }
-
     private const int MIN_SIZE = 1;
     private const int MIN_POS = 10;
     private const int GRID_THRESHOLD = 3;
 
     public signal void canvas_moved (double delta_x, double delta_y);
     public signal void canvas_scroll_set_origin (double origin_x, double origin_y);
-
+    public Akira.Models.ListModel<Lib.Items.CanvasArtboard> artboards;
```

Comment:
Why are you declaring this here?
You should get this list from the ItemsManager.